spec/KeyValueAdapter.spec.ts

```diff
@@ -16,4 +16,20 @@ describe('KeyValueAdapter', () => {
         expect(adapter.adapt('test')).toBe('hi');
         expect(adapter.adapt('value')).toBe('world');
     });
+
+    it('should error', () => {
+        interface ITestMap {
+            test: string;
+            value: string;
+        }
+
+        let adapter: KeyValueAdapter<ITestMap> = new KeyValueAdapter({
+            test: "hi",
+            value: "world"
+        });
+
+        expect(() => {
+            adapter.adapt(<any>"asdf");
+        }).toThrowError(/doesn't exist in map\.$/);
```

Comment:
```suggestion
        let key = "asdf";
        expect(() => {
            adapter.adapt(<any>key);
        }).toThrowError(`Key "${key}" doesn't exist in map.`);
```

Reply:
"doesn't exist in map" is not a clear message. Enforce the need to include faulty key in the error message.